InnoGotchiBackend/InnnoGotchi.DAL/Respositories/PetRepository.cs

```diff
@@ -4,42 +4,65 @@
 
 namespace InnnoGotchi.DAL.Respositories
 {
-    public class PetRepository : InnoGotchiContextHandler, IRepository<Pet>
+    /// <summary>
+    /// Represents repository that has full access to pet entities in database
+    /// </summary>
+    public class PetRepository : IRepository<Pet>
     {
-        public PetRepository(InnoGotchiContext context) : base(context) { }
+        private InnoGotchiContext _context;
+        public PetRepository(InnoGotchiContext context)
+        {
+            _context = context;
+        }
+
+        public bool Contains(Func<Pet, bool> predicate)
+        {
+            Pet? pet = FirstOrDefault(predicate);
+            if (pet == null)
+                return false;
+            else
+                return true;
+        }
 
         public void Create(Pet item)
         {
             _context.Pets.Add(item);
         }
 
-        public void Delete(int id)
+        public bool Delete(int id)
         {
-            Pet? pet = _context.Pets.FirstOrDefault(p => p.Id == id);
+            Pet? pet = Get(id);
             if (pet != null)
             {
                 _context.Pets.Remove(pet);
+                return true;
             }
+            return false;
+        }
+
+        public IQueryable<Pet> FindAll(Func<Pet, bool> predicate)
+        {
+            return GetAll().Where(predicate).AsQueryable();
         }
 
-        public IEnumerable<Pet> Find(Func<Pet, bool> predicate)
+        public Pet? FirstOrDefault(Func<Pet, bool> predicate)
         {
-            return _context.Pets.Where(predicate);
+            return GetAll().FirstOrDefault(predicate);
         }
 
         public Pet? Get(int id)
         {
-            return _context.Pets.FirstOrDefault(p => p.Id == id);
+            return FirstOrDefault(p => p.Id == id);
         }
 
-        public IEnumerable<Pet> GetAll()
+        public IQueryable<Pet> GetAll()
         {
             return _context.Pets;
         }
 
-        public void Update(int id, Pet item)
+        public void Update(Pet item)
         {
-            item.Id = id;
+            _context.ChangeTracker.Clear();
```

Comment:
Bad practice...

Reply:
How can I fix that, because without this throwing this exception
![image](https://user-images.githubusercontent.com/74291930/216837158-99a45da6-fb13-4b4a-a245-e4d2d58a3154.png)
I can do something like this 
![image](https://user-images.githubusercontent.com/74291930/216837437-b9f0ed88-ac33-45ad-9c1c-cb1321f77358.png)
But in this case it's not working and I am already not sure that this solution is right

Reply:
Yes, try to avoid change tracker without cheating like this. You make EF unhappy, when trying to hack him by this. 

Currently you want to use this approach: you changed object, but say EF that you didn't change it.